File: WeatherServer/WeatherServer.API/Models/Astro.cs
using System.Text.Json.Serialization;

namespace WeatherServer.API.Models
{
    public record Astro(
            string sunrise,
            string sunset,
            string moonrise,
            string moonset,
            [property: JsonPropertyName("moon_phase")] string moonPhase,
            [property: JsonPropertyName("moon_illumination")] double moonIllumination,
            [property: JsonPropertyName("is_moon_up")] int isMoonUp,
            [property: JsonPropertyName("is_sun_up")] int isSunUp
        );
}

File: WeatherFlutter/weather_flutter/lib/model/astro.dart
import 'package:json_annotation/json_annotation.dart';

part 'astro.g.dart';

@JsonSerializable()
class Astro {
  final String sunrise;
  final String sunset;
  final String moonrise;
  final String moonset;

  @JsonKey(name: 'moon_phase')
  final String moonPhase;

  @JsonKey(name: 'moon_illumination')
  final double moonIllumination;

  @JsonKey(name: 'is_moon_up')
  final int isMoonUp;

  @JsonKey(name: 'is_sun_up')
  final int isSunUp;

  Astro({
    required this.sunrise,
    required this.sunset,
    required this.moonrise,
    required this.moonset,
    required this.moonPhase,
    required this.moonIllumination,
    required this.isMoonUp,
    required this.isSunUp,
  });

  factory Astro.fromJson(Map<String, dynamic> json) => _$AstroFromJson(json);
}
